lib/types/index.d.ts

```diff
@@ -807,23 +807,23 @@ export class Linter {
 
 	verify(
 		code: SourceCode | string,
-		config: Linter.LegacyConfig | Linter.Config | Linter.Config[],
+		config: Linter.Config | Linter.Config[],
 		filename?: string,
 	): Linter.LintMessage[];
 	verify(
 		code: SourceCode | string,
-		config: Linter.LegacyConfig | Linter.Config | Linter.Config[],
+		config: Linter.Config | Linter.Config[],
 		options: Linter.LintOptions,
 	): Linter.LintMessage[];
 
 	verifyAndFix(
 		code: string,
-		config: Linter.LegacyConfig | Linter.Config | Linter.Config[],
+		config: Linter.Config | Linter.Config[],
 		filename?: string,
```

Comment:
I'm not sure if the intent was to allow a `string` as a shortcut for `options.filename` here, as it is clearly the case for the `verify()` method. The logic is a bit inconsistent and this case is not tested. Also, the usage is undocumented (for both `verify()` and `verifyAndFix()`).